### DISCUSSION TITLE
PickingHitArea - can renderer param be optional?

### DISCUSSION THREAD

**Discussion by goldenratio** [jnsmalm/pixi3d#86] [Ideas]
Hi,

Was just wondering, since pixi3d has reference to pixi's Renderer, can we have default value https://github.com/jnsmalm/pixi3d/blob/develop/src/picking/picking-hitarea.ts#L24 for renderer param? might be a bad idea, I guess. Since it's a breaking API. Would like to hear your thoughts?
```
constructor(public object: Mesh3D | Model, renderer: Renderer = Camera.main.renderer, public camera?: Camera) {
    this._picking = renderer.plugins.picking
  }
```

**Comment by jnsmalm** [comment]
Yes, this would have been better but I'm not too keen on changing it at this point because as you already said it breaks the API.

**Comment by jnsmalm** [comment]
You need this for a specific reason or only convienience?

**Reply by goldenratio** [reply]
just for convienience

**Reply by goldenratio** [reply]
sorry to bug you in this again :D 
maybe a static function inside PickingHitArea ? I can make a PR
``` 
static fromObject(object: Mesh3D | Model): PickingHitArea {
    return new PickingHitArea(Camera.main.renderer, object)
  }
```

**Reply by jnsmalm** [reply]
Please create a PR.
